### PR TITLE
Added Jest/Enzyme.

### PR DESCRIPTION
**Pull request by Sam0064**
Added Jest/Enzyme, added a README file with directions as to how I implemented the packages, how I set them up in the configuration files, and some troubleshooting options should the build fail.

### CODE REVIEW

**Review by lazdinst — APPROVED** [review]
Talis Lazdins - Approved.

**Review by kevinzhuang10 — APPROVED** [review]
Good Stuff

File: sum.test.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const sum = require('./sum');
```
Comment (olegtrofimov1) [inline sum.test.js:1]: This looks like something from development that should probably not be in the main codebase.

File: server/test/server.spec.js [hunk 2]
```diff
@@ -3,9 +3,10 @@ const request = require('supertest');
 const express = require('express');
 const expect = require('chai').expect;
 const app = require('../app.js');
+global.Promise = require.requireActual('promise'); //ANY DB tests MUST have this line of code!!
 
 describe('basic server', function() {
-  it('sends back hello world', function(done) {
+  xit('sends back hello world', function(done) {
```
Comment (olegtrofimov1) [inline server/test/server.spec.js:9]: Why did these user and profile tests get commented out?

File: server/middleware/passport.js [hunk 3]
```diff
@@ -32,107 +32,107 @@ passport.use('local-signup', new LocalStrategy({
...
 
-passport.use('facebook', new FacebookStrategy({
-  clientID: config.Facebook.clientID,
-  clientSecret: config.Facebook.clientSecret,
-  callbackURL: config.Facebook.callbackURL,
-  profileFields: ['id', 'emails', 'name']
-},
-  (accessToken, refreshToken, profile, done) => getOrCreateOAuthProfile('facebook', profile, done))
-);
+// passport.use('facebook', new FacebookStrategy({
```
Comment (olegtrofimov1) [inline server/middleware/passport.js:118]: Please delete code that you are not using. Do not leave commented out code.

File: client/test/TestingREADME.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
+#Intro
```
Comment (olegtrofimov1) [inline client/test/TestingREADME.md:1]: This is fantastic!!!!

File: client/test/TestingREADME.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+TESTING IS FRAGILE. I have listed every change that I made below. Stuff will break, if you add/change stuff in our testing configuration, please document it below, as it may need modified in the future. 
+
+We are going to use Jest and Enzyme. 
+Jest is the actual Facebook testing suite for React, and allows us to take snapshot tests to ensure visual consistency when we are building our application. 
+
+Enzyme is used to test React Components. It is good for doing things such as ensuring that Component lifecyle methods are rendering correctly, ensuring that components are loading in the proper fashion, and ensuring that non-visual events such as click or on change events are properly firing. 
+
+Both of these packages will be set up to run via npm test. 
+
+sum.js && sum.test.js are just boilerplate tests to ensure that Jest is running.
```
Comment (olegtrofimov1) [inline client/test/TestingREADME.md:18]: Why do you need things that test if Jest is running? Are these two files really something you need?

**Pull request merged**